### ISSUE TITLE
Scenario#log(..) with PrettyFormatter has problems with technical log output - like URLEncoded values.

### ISSUE THREAD

**Issue by skloessel** [cucumber/cucumber-jvm#2544]
Hello,

I use Scenario.log(String) to log some technical parameters e.g. URLs in cucumber scenarios.

Since the latest version 7.3.3. this leads to problems, because the PrettyFormatter applies String.format(...) to the message to be logged and therefore the string is interpreted as a format message. 

Example log-message:
`https://localhost:35403/oje/oidc/client-authorization-debug?init_client_from=selenium&init_client_id=jwebclient&init_client_secret=jwebclientsecret&init_scope=openid+email&init_state=DcIFM&init_nonce=QSqUyzfp4m&init_claims=%7B%22id_token%22%3A%7B%22name%22%3Anull%7D%7D&init_response_type=id_token+token
`

Code-Snippet from PrettyFormatter:

```    
private void printText(WriteEvent event) {
        // Prevent interleaving when multiple threads write to System.out
        StringBuilder builder = new StringBuilder();
        try (BufferedReader lines = new BufferedReader(new StringReader(event.getText()))) {
            String line;
            while ((line = lines.readLine()) != null) {
                builder.append(String.format(STEP_SCENARIO_INDENT + line + "%n"));
            }
        } catch (IOException e) {
            throw new CucumberException(e);
        }
        out.append(builder);
 }
```

The String.format(..) statement in the line `builder.append(String.format(STEP_SCENARIO_INDENT + line + "%n"));`  doesn't make sense to me since no parameters are passed there.

### ✅ What did you expect to see?

+ The PrettyFormatter should reliably handle log messages that include technical encodings - such as URLEncoded strings.
+ These messages should not undergo further unintended formatting.

### Stacktrace

```
[WARN ] 2022-05-02 10:36:29.923 [main] WithScenarioUtils - error logging message: 
User navigates to the client uri 'https://localhost:35403/oje/oidc/client-authorization-debug?init_client_from=selenium&init_client_id=jwebclient&init_client_secret=jwebclientsecret&init_scope=openid+email&init_state=DcIFM&init_nonce=QSqUyzfp4m&init_claims=%7B%22id_token%22%3A%7B%22name%22%3Anull%7D%7D&init_response_type=id_token+token'
java.util.UnknownFormatConversionException: Conversion = 'i'
	at java.util.Formatter$FormatSpecifier.conversion(Formatter.java:2935) ~[?:?]
	at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2971) ~[?:?]
	at java.util.Formatter.parse(Formatter.java:2827) ~[?:?]
	at java.util.Formatter.format(Formatter.java:2751) ~[?:?]
	at java.util.Formatter.format(Formatter.java:2705) ~[?:?]
	at java.lang.String.format(String.java:4138) ~[?:?]
	at io.cucumber.core.plugin.PrettyFormatter.printText(PrettyFormatter.java:189) ~[cucumber-core-7.3.3.jar:7.3.3]
	at io.cucumber.core.plugin.PrettyFormatter.handleWrite(PrettyFormatter.java:78) ~[cucumber-core-7.3.3.jar:7.3.3]
	at io.cucumber.core.eventbus.AbstractEventPublisher.send(AbstractEventPublisher.java:51) ~[cucumber-core-7.3.3.jar:7.3.3]
	at io.cucumber.core.eventbus.AbstractEventBus.send(AbstractEventBus.java:12) ~[cucumber-core-7.3.3.jar:7.3.3]
	at io.cucumber.core.runner.TestCaseState.log(TestCaseState.java:107) ~[cucumber-core-7.3.3.jar:7.3.3]
	at io.cucumber.java.Scenario.log(Scenario.java:100) ~[cucumber-java-7.3.3.jar:7.3.3]
```

### Test-Case (String.format ... only)

The following snippet demonstrates that String.format(...) is not suitable to log URLEncoded messages.
At position 263 '...%22i...' becomes recognized as a potential numeric argument. However, no parameters are available.

```
class PrettyFormatterTest {
	
	final String MESSAGE = "User navigates to the client uri 'https://localhost:35403/oje/oidc/client-authorization-debug?init_client_from=selenium&init_client_id=jwebclient&init_client_secret=jwebclientsecret&init_scope=openid+email&init_state=DcIFM&init_nonce=QSqUyzfp4m&init_claims=%7B%22id_token%22%3A%7B%22name%22%3Anull%7D%7D&init_response_type=id_token+token'";

	@Test
	void stringFormatFailsTest() {
		
		String.format(MESSAGE);
	}

}
```

**Comment by mpkorstanje** [comment]
> The String.format(..) statement in the line builder.append(String.format(STEP_SCENARIO_INDENT + line + "%n")); doesn't make sense to me since no parameters are passed there.

Yeah. I reckon The `%n` can be replaced with `System.newLine()`. Feel free to send a PR.

**Comment by mpkorstanje** [comment]
Fixed by #2545
